File: ArchLucid.Decisioning/Findings/ActorDependentFindingEngineTypes.cs
namespace ArchLucid.Decisioning.Findings;

/// <summary>Built-in engines that require <c>GraphNodeTypes.Actor</c> nodes to produce findings (DX-03 / TB-2344).</summary>
public static class ActorDependentFindingEngineTypes
{
    public const string ExternalExposure = "external-exposure";

    public const string TrustBoundary = "trust-boundary";

    public const string PrivilegedAccess = "privileged-access";

    public static IReadOnlyList<string> All { get; } =
    [
        ExternalExposure,
        TrustBoundary,
        PrivilegedAccess,
    ];
}

File: archlucid-ui/src/lib/findings/actor-dependent-finding-engine-types.ts
/** Built-in engine types that require Actor nodes (DX-03 / DX-15). */
export const ACTOR_DEPENDENT_FINDING_ENGINE_TYPES = [
  "external-exposure",
  "trust-boundary",
  "privileged-access",
] as const;

export type ActorDependentFindingEngineType = (typeof ACTOR_DEPENDENT_FINDING_ENGINE_TYPES)[number];
